pkg/capi/gcp.go

```diff
@@ -0,0 +1,308 @@
+package capi
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+	framework "github.com/openshift/cluster-api-actuator-pkg/pkg/framework"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gcpv1 "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/komega"
+	yaml "sigs.k8s.io/yaml"
+)
+
+const (
+	gcpMachineTemplateName     = "gcp-machine-template"
```

Comment:
Perhaps you want to use `generateName` to avoid AlreadyExists errors, and then you definitely have a unique template for each test case

Reply:
Thanks , will included this , it definitely will avoid the issue of template with same name issue.

Reply:
@miyadav I think Joel here meant to use ObjectMeta's `GenerateName` inside the `createdTemplate := &gcpv1.GCPMachineTemplate{GenerateName: <PREFIX> }` declaration, and then use the `client.ObjectKeyFromObject(createdTemplate)`, you can see it in action here: https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/a611e0dbd02dcde40b5e4faec47630be298e3ece/pkg/webhooks/controlplanemachineset/webhooks.go#L332

Reply:
Ah , got you , thanks ..